File: C/old_approach/old_approach.c
#include <stdio.h>

int reverse_num(int number)
{
	int num_return = 0;
	int digit = 0;
	int i = 0;
	while (number != 0)
	{
		digit = number % 10;
		num_return = num_return * 10 + digit;
		number /= 10;
	}
	return num_return;
}

int main(int argc, const char * argv[])
{
	int number = 0;
	int rc = fscanf(stdin, "%d", &number);
	if (rc < 0) {
		printf("Please enter a number: ");
	}

	int res = reverse_num(number);
	printf("Result: %d\n", res);
	return 0;
}

File: Perl/old_approach/old_approach.pl
sub reverse_num
{
	my $num = shift;
	my $sum = 0;
	my $digit = 0;

	while ($num != 0)
	{
		$digit = $num % 10;
		$sum = $sum * 10 + $digit;
		$num = int($num / 10);
	}

	return $sum;
}

print "Please insert a number: ";
$input = <STDIN>;
$number = int($input);
$res = reverse_num($number);
print $res;